pkg/cloudprovider/aws/validation.go

```diff
@@ -36,10 +38,17 @@ func (c *Capacity) validateLabels() error {
 				if !functional.ContainsString(capacityTypes, value) {
 					return fmt.Errorf("%s must be one of %v", key, capacityTypes)
 				}
+			case launchTemplateIdLabel:
```

Comment:
Not a fan of defining booleans, mutating them, and then checking them after the switch. This logic will quickly get out of hand as we add more complexity. I was wary about using a switch statement for validation for this exact reason. 

Consider 
```
f.ValidateAll(
 func() error { validateCapacityLabel(labels) },
 func() error { validateLaunchTemplateLabel(labels) },
)
```
If no label, simply pass the check.

Reply:
Aye, aye, cap'n!